applications/leaderboard/Leaderboard/Program.cs

```diff
@@ -0,0 +1,525 @@
+                                cmd.Parameters["PlayerId"].Value = reader.GetFieldValue<int>("PlayerId");
+                                // Insert random value for score between 10000 and 1000000.
+                                cmd.Parameters["Score"].Value = r.Next(1000, 1000001);
+                                // Insert random past timestamp value into Timestamp column.
+                                cmd.Parameters["Timestamp"].Value = randomTimestamp.ToString("o");
+                                cmd.ExecuteNonQuery();
+                            }
+                            await tx.CommitAsync();
+                        }
+                    }
+                }
+            }
+            Console.WriteLine("Done inserting score records...");
+        }
+
+        public static object QueryTopTenAllTime(string projectId,
+            string instanceId, string databaseId)
+        {
+            var response = QueryTopTenAllTimeAsync(
+                projectId, instanceId, databaseId);
+            response.Wait();
+            return ExitCode.Success;
+        }
+
+        public static async Task QueryTopTenAllTimeAsync(
+            string projectId, string instanceId, string databaseId)
+        {
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/"
+            + $"{instanceId}/databases/{databaseId}";
+            // Create connection to Cloud Spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd = connection.CreateSelectCommand(
+                    @"SELECT p.PlayerId, p.PlayerName, s.Score, s.Timestamp
+                        FROM Players p
+                        JOIN Scores s ON p.PlayerId = s.PlayerId
+                        ORDER BY s.Score DESC LIMIT 10");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine("PlayerId : "
+                            + reader.GetFieldValue<string>("PlayerId")
+                            + " PlayerName : "
+                            + reader.GetFieldValue<string>("PlayerName")
+                            + " Score : "
+                            + reader.GetFieldValue<string>("Score")
+                            + " Timestamp : "
+                            + reader.GetFieldValue<string>("Timestamp"));
+                    }
+                }
+            }
+        }
+
+        public static object QueryTopTenOfYear(string projectId,
+            string instanceId, string databaseId)
+        {
+            var response = QueryTopTenOfYearAsync(
+                projectId, instanceId, databaseId);
+            response.Wait();
+            return ExitCode.Success;
+        }
+
+        public static async Task QueryTopTenOfYearAsync(
```

Comment:
OfYear, OfWeek, OfMonth....  I think these 3 functions can be written as one function with one more parameter, the time span.